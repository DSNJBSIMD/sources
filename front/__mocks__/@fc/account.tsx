import type { PropsWithChildren } from 'react';
import React from 'react';

export const AccountContext = React.createContext(expect.any(Object));

export const AccountProvider = jest.fn(
  ({ children, config }: { config: unknown } & PropsWithChildren) => (
    <AccountContext.Provider value={config}>
      <div data-mockid="AccountProvider">
        <div>{children}</div>
      </div>
    </AccountContext.Provider>
  ),
);

export const ConnectValidator = {
  validate: jest.fn(),
};

export const Options = {
  CONFIG_NAME: 'Account',
};
