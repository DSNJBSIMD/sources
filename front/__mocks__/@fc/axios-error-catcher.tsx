import type { PropsWithChildren } from 'react';
import React from 'react';

export const AxiosErrorCatcherContext = React.createContext(expect.any(Object));

export const AxiosErrorCatcherProvider = jest.fn(
  ({ children, value }: { value: unknown } & PropsWithChildren) => (
    <AxiosErrorCatcherContext.Provider value={value}>{children}</AxiosErrorCatcherContext.Provider>
  ),
);
