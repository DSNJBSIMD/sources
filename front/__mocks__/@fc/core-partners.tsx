export const CreateInstanceButton = jest.fn(() => <div data-mockid="CreateInstanceButton" />);

export const InstanceComponent = jest.fn(() => <div data-mockid="InstanceComponent" />);

export const InstancesListComponent = jest.fn(() => <div data-mockid="InstancesListComponent" />);

export const PartnersService = {
  commit: jest.fn(),
  get: jest.fn(),
};

export enum Environment {
  SANDBOX = 'SANDBOX',
  PRODUCTION = 'PRODUCTION',
}

export enum Options {
  CONFIG_NAME = 'Partners',
}

export enum PublicationStatus {
  DRAFT = 'DRAFT',
  PENDING = 'PENDING',
  PUBLISHED = 'PUBLISHED',
}
