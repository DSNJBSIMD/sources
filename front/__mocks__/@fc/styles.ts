import type { ReactElement } from 'react';
import React from 'react';

import type { StylesProviderProps } from '@fc/styles';

export const useStylesQuery = jest.fn();

export const useStylesContext = jest.fn();

export const useStylesVariables = jest.fn(() => []);

export const StylesProvider = jest.fn(
  ({ children }: StylesProviderProps) => children as ReactElement,
);

export const StylesContext = React.createContext(expect.any(Object));

export enum Strings {
  DOUBLE_DASH = '--',
}
