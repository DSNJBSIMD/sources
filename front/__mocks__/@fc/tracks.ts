export const TracksListComponent = jest.fn();

export enum Options {
  CONFIG_NAME = 'Tracks',
}
