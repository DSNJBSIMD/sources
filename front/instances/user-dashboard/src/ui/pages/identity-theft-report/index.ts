export * from './identity-theft-report.page';
