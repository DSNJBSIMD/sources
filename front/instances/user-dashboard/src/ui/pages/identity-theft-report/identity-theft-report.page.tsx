import { Helmet } from '@dr.pogodin/react-helmet';
import classnames from 'classnames';
import React from 'react';

import { useStylesQuery, useStylesVariables } from '@fc/styles';

export const IdentityTheftReportPage = React.memo(() => {
  const [breakpointLg] = useStylesVariables(['breakpoint-lg']);
  const gtDesktop = useStylesQuery({ minWidth: breakpointLg });

  return (
    <React.Fragment>
      <Helmet>
        <title>Mon tableau de bord - Signaler une usurpation</title>
      </Helmet>
      <div
        className={classnames('fr-m-auto fr-px-2w', {
          // Class CSS
          // eslint-disable-next-line @typescript-eslint/naming-convention
          'fr-mt-5w': !gtDesktop,
          // Class CSS
          // eslint-disable-next-line @typescript-eslint/naming-convention
          'fr-mt-8w': gtDesktop,
        })}
        id="page-container">
        <h1>Fishing Report</h1>
        <p>This is the fishing report page.</p>
        <p>Hello World !</p>
      </div>
    </React.Fragment>
  );
});

IdentityTheftReportPage.displayName = 'IdentityTheftReportPage';
