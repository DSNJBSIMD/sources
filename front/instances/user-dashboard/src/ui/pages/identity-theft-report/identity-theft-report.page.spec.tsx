import { render } from '@testing-library/react';

import { useStylesQuery, useStylesVariables } from '@fc/styles';

import { IdentityTheftReportPage } from './identity-theft-report.page';

describe('IdentityTheftReportPage', () => {
  // Given
  const cssVariablesMock = new CSSStyleDeclaration();
  cssVariablesMock.setProperty('--breakpoint-lg', 'any-pixel-value');

  const getComputedStyleMock = jest.spyOn(window, 'getComputedStyle');
  jest.mocked(getComputedStyleMock).mockReturnValue(cssVariablesMock);

  beforeEach(() => {
    // @NOTE used to prevent useStylesVariables.useStylesContext to throw
    // useStylesContext requires to be into a StylesProvider context
    jest.mocked(useStylesVariables).mockReturnValue(['any-pixel-value']);
  });

  it('should call the responsive hooks, useStylesVariables and useStylesQuery', () => {
    // When
    render(<IdentityTheftReportPage />);

    // Then
    expect(useStylesVariables).toHaveBeenNthCalledWith(1, ['breakpoint-lg']);
    expect(useStylesQuery).toHaveBeenNthCalledWith(1, { minWidth: 'any-pixel-value' });
  });

  it('should match the snapshot, display into a desktop viewport', () => {
    // Given
    jest.mocked(useStylesQuery).mockReturnValueOnce(true);

    // When
    const { container } = render(<IdentityTheftReportPage />);

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot, display into a mobile viewport', () => {
    // Given
    jest.mocked(useStylesQuery).mockReturnValueOnce(false);

    // When
    const { container } = render(<IdentityTheftReportPage />);

    // Then
    expect(container).toMatchSnapshot();
  });
});
