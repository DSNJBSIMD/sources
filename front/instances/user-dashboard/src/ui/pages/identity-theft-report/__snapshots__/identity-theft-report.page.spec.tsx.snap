// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdentityTheftReportPage should match the snapshot, display into a desktop viewport 1`] = `
<div>
  <div
    data-mockid="Helmet"
  >
    <span
      data-htmltag="title"
    >
      Mon tableau de bord - Signaler une usurpation
    </span>
  </div>
  <div
    class="fr-m-auto fr-px-2w fr-mt-8w"
    id="page-container"
  >
    <h1>
      Fishing Report
    </h1>
    <p>
      This is the fishing report page.
    </p>
    <p>
      Hello World !
    </p>
  </div>
</div>
`;

exports[`IdentityTheftReportPage should match the snapshot, display into a mobile viewport 1`] = `
<div>
  <div
    data-mockid="Helmet"
  >
    <span
      data-htmltag="title"
    >
      Mon tableau de bord - Signaler une usurpation
    </span>
  </div>
  <div
    class="fr-m-auto fr-px-2w fr-mt-5w"
    id="page-container"
  >
    <h1>
      Fishing Report
    </h1>
    <p>
      This is the fishing report page.
    </p>
    <p>
      Hello World !
    </p>
  </div>
</div>
`;
